### PR TITLE
143539949 ne create lot

### PR DESCRIPTION
**Pull request by ski-climb**
@tmikeschu @ethanbennett @DavidKnott Ready for review.

This PR is basically creating a Lot via the API.  A `product_id` needs to be submitted as well because a lot `belongs_to` a product, so this PR also includes the `index` action (via `/api/v1/products`) to list the products so that the `product_id` can be submitted as part of the form which creates a new Lot.

There will be some merge conflicts once my other PR is merged, and I'm completely happy to sort those out.

This PR was a bit slimmer than I was expecting.  Please let me know if this doesn't match up with the way you might expect the React front end to work/submit information - happy to make changes.

[(Closes) #143539949]

### CODE REVIEW

File: app/controllers/api/v1/lots_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      render json: lot, status: 201
+    else
+      head 400
+    end
+  end
+
+  private
+
+    def product_id
+      params[:lot][:product_id]
```
Comment (tmikeschu) [inline app/controllers/api/v1/lots_controller.rb:16]: I think this ultimately has the same effect as strong params, but any reason to not use them?

File: spec/requests/api/v1/lots_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        lot: {
+          product_id: Product.first.id
+        }
+      }
+    }.to change {Lot.count}.by 1
+
+    lot = JSON.parse(response.body)
+
+    expect(response.status).to eq 201
+    expect(lot['id']).to be_an Integer
```
Comment (tmikeschu) [inline spec/requests/api/v1/lots_spec.rb]: Could we add something like `expect(lot['product_id']).to eq Product.first.id` ?

**Review by tmikeschu — CHANGES_REQUESTED** [review]
Looks great! One request on the spec

File: spec/requests/api/v1/lots_spec.rb (hunk 2 above)
Comment (ski-climb) [reply spec/requests/api/v1/lots_spec.rb]: @tmikeschu Thanks for the suggestion - I'll add that.

File: app/controllers/api/v1/lots_controller.rb (hunk 1 above)
Comment (ski-climb) [reply app/controllers/api/v1/lots_controller.rb:16]: @tmikeschu Good eye.  I tried strong params at first, but that returns a params hash, whereas here we're using the `id` to find a particular product.

**Review by tmikeschu — APPROVED** [review]

File: spec/requests/api/v1/lots_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  it "can create a lot" do
+    expect { 
+      post "/api/v1/lots", params: {
+        lot: {
+          product_id: Product.first.id
+        }
+      }
+    }.to change {Lot.count}.by 1
+
```
Comment (DavidKnott) [inline spec/requests/api/v1/lots_spec.rb:14]: Nice job using change!

**Review by DavidKnott — APPROVED** [review]

**Pull request merged**